spec/features/saml/loa3_sso_spec.rb

```diff
@@ -33,6 +33,22 @@
     end
   end
 
+  context 'canceling verification', js: true do
+    it 'returns user to profile page if they have previously signed up' do
+      sign_in_and_2fa_user
+
+      loa3_sp_session do |session|
+        session['warden.user.user.session'][:first_time_recovery_code_view] = true
+      end
+
+      visit verify_path
+
+      click_on t('links.cancel')
+      click_on t('loa3.buttons.cancel')
+      expect(current_url).to match(/profile/)
```

Comment:
Searching thru the code for this key, I think the reason the behavior is backwards is because I'm using the key backwards. It looks like it gets set when the user has signed up but hasn't yet see the recovery code screen.